src/main/java/org/apache/mesos/specification/DefaultTaskSpecification.java

```diff
@@ -19,24 +21,33 @@
     private final Protos.CommandInfo commandInfo;
     private final Collection<ResourceSpecification> resourceSpecifications;
     private final Collection<VolumeSpecification> volumeSpecifications;
+    private final Optional<PlacementRuleGenerator> placementOptional;
 
     public static DefaultTaskSpecification create(Protos.TaskInfo taskInfo) throws InvalidTaskSpecificationException {
+        //TODO(nick): Is the original placement constraint needed here?
```

Comment:
Anyone have any comments on this? It looked like we should be fine in practice but I wasn't certain. Should TaskSpecifications be derivable from TaskInfos in the first place?